### ISSUE TITLE
Relative container needs renderedPosition() on each popper

### ISSUE THREAD

**Issue by jelmerjellema** [cytoscape/cytoscape.js-popper#56]
We use a popup menu and an always-on small question mark through popper. The cytoscape canvas is in a relative positioned container (because of absolute placement of other element). 

Popper placement is always exactly (x,y) off position, where (x,y) is the position of the cytoscape container. 

So each element.popper(...) call needs a call to a "fix position" function like this:
```
 renderedPosition: el => {
          let clpos = this.clientpos();
          let elpos = el.renderedPosition();

          return {
            x: elpos.x - clpos.x,
            y: elpos.y - clpos.y
          };
        }
```
Where this.clientpos() is a function in the angular component that encapsulates cytoscape:
```
  private clientpos(el = this.canvas.nativeElement): { x: number, y: number } {
    let x = 0;
    let y = 0;

    while (el) {
      x += (el.offsetLeft - el.scrollLeft + el.clientLeft);
      y += (el.offsetTop - el.scrollTop + el.clientTop);
      el = el.offsetParent;
    }
    return {x: x, y: y};
  }
```

Maybe this could be fixed by using the client position of the position container in the calculations?

Regards, Jelmer

**Comment by jelmerjellema** [comment]
The code above would give the centered position, but I found out through the source code what we need is the left-top, so I now use renderedBoundingBox().x1 and .y1.

**Comment by maxkfranz** [comment]
Have you investigated or enquired with the Popper documentation or issue tracker?

[This wrapper already takes into account the container's position](https://github.com/cytoscape/cytoscape.js-popper/blob/aba6c2490c3fd1669118cb6f97a96ae66549b6fc/src/bb.js#L3), so there may be something you need to investigate on the Popper side.

**Comment by jelmerjellema** [comment]
Hi Max,

Yes, I did. And it all works fine with the canvas in the top left corner. But I guess it will be popper doing a wrong transform / absolute placement without moving the dom element to the body. 

It think it happens because of a relative placement of the canvas. The code at https://github.com/cytoscape/cytoscape.js-popper/blob/aba6c2490c3fd1669118cb6f97a96ae66549b6fc/src/bb.js#L3 does not consider the first positioned container, and will come up with absolute coordinates, while the popper is in a relative container.

I could move the popper outside of the container, but running in Angular and all, this means moving it out of a reusable component. 

For now, I fixed this by making a non popper popup component in angular for one of the two poppers,and using the code in my post above for the other one.

Jelmer
